packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { EmailHeader, EmailType } from '../../lib/email';
+import { Page, expect, test } from '../../lib/fixtures/standard';
+import { BaseTarget } from '../../lib/targets/base';
+import { SettingsPage } from '../../pages/settings';
+import { DeleteAccountPage } from '../../pages/settings/deleteAccount';
+
+const AGE_21 = '21';
+
+/**
+ * These tests represent various permutations between interacting with V1 and V2
+ * key stretched passwords. We need to ensure that operations are interchangeable!
+ */
+test.describe('severity-2 #smoke', () => {
+  type Version = { version: 1 | 2; query: string };
+  type TestCase = { signup: Version; signin: Version };
+  const v1: Version = { version: 1, query: '' };
+  const v2: Version = { version: 2, query: 'stretch=2' };
```

Comment:
For (Part 3), i wonder if we should export these so we can share them between files:
```sh
git grep -Pn "type (TestCase|Version) = " packages | cat

packages/functional-tests/tests/key-stretching-v2/changePassword.spec.ts:31:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/changePassword.spec.ts:32:  type TestCase = { signup: Version; change: Version };
packages/functional-tests/tests/key-stretching-v2/recoveryKey.spec.ts:35:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/recoveryKey.spec.ts:36:  type TestCase = { signup: Version; reset: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/resetPassword.spec.ts:34:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/resetPassword.spec.ts:35:  type TestCase = { signup: Version; reset: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:18:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:19:  type TestCase = { signup: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/totp.spec.ts:32:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/totp.spec.ts:33:  type TestCase = { signup: Version; signin: Version };
```

&hellip; aaactually, it looks like `TestCase` can be different in some files, so only `Version` could be exported. Wonder if it's even worth it.

Reply:
I'm inclined to leave them honestly. It's verbose, but I think easier to understand if we define them in the test module.